pkg/client/new_extension.go

```diff
@@ -0,0 +1,91 @@
+package client
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
+
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/pkg/dist"
+)
+
+type NewExtensionOptions struct {
+	// api compat version of the output extension artifact.
+	API string
+
+	// The base directory to generate assets
+	Path string
+
+	// The ID of the output extension artifact.
+	ID string
+
+	// version of the output extension artifact.
+	Version string
+
+	// Deprecated: The stacks this extension will work with
+	Stacks []dist.Stack
+}
+
+func (c *Client) NewExtension(ctx context.Context, opts NewExtensionOptions) error {
+	err := createExtensionTOML(opts.Path, opts.ID, opts.Version, opts.API, c)
+	if err != nil {
+		return err
+	}
+	return createBashExtension(opts.Path, c)
+}
+
+func createBashExtension(path string, c *Client) error {
+	if err := createBinScript(path, "build", bashBinBuild, c); err != nil {
```

Comment:
```suggestion
	if err := createBinScript(path, "generate", bashBinGenerate, c); err != nil {
```

Reply:
We could do something like the below for `bashBinGenerate`:

```
#!/usr/bin/env bash
set -eo pipefail

# 1. GET ARGS
output_dir=$CNB_OUTPUT_DIR

# 2. GENERATE build.Dockerfile
cat >> "${output_dir}/build.Dockerfile" <<EOL
ARG base_image
FROM \${base_image}

RUN echo "Hello from build extension"
EOL

# 3. GENERATE run.Dockerfile
cat >> "${output_dir}/run.Dockerfile" <<EOL
ARG base_image
FROM \${base_image}

RUN echo "Hello from run extension"
EOL
```

Reply:
https://buildpacks.io/docs/extension-guide/create-extension/ contains more information for the curious...